src/client/Microsoft.Identity.Client/ApiConfig/Executors/PublicClientExecutor.cs

```diff
@@ -28,22 +28,18 @@ public PublicClientExecutor(IServiceBundle serviceBundle, PublicClientApplicatio
         {
             var requestContext = CreateRequestContextAndLogVersionInfo(commonParameters.CorrelationId);
 
-            var requestParams = _publicClientApplication.CreateRequestParameters(
+            AuthenticationRequestParameters requestParams = _publicClientApplication.CreateRequestParameters(
                 commonParameters,
                 requestContext,
                 _publicClientApplication.UserTokenCacheInternal);
 
             requestParams.LoginHint = interactiveParameters.LoginHint;
             requestParams.Account = interactiveParameters.Account;
 
-            var handler = new InteractiveRequest(
-                ServiceBundle,
-                requestParams,
-                interactiveParameters,
-                CreateWebAuthenticationDialog(interactiveParameters, requestParams.RequestContext));
-
-            return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
+            InteractiveRequest interactiveRequest = 
+                new InteractiveRequest(requestParams, interactiveParameters);
```

Comment:
So beautiful.  They should have sent a poet.

Reply:
Dear external contributor, thank you for your contribution. I am finding it hard to understand how this can be improved. 😃

Reply:
The whole executor logic and params objects that you introduced are epic Mark, makes things so nice :)

Reply:
@MarkZuber ...lol...